eval/service.py

```diff
@@ -0,0 +1,695 @@
+				Online_Mind2Web_eval_with_retry(task_description, action_history, screenshot_paths, model, score_threshold)
+			)
+
+			if eval_result is None:
+				raise Exception('Evaluation failed after all retries')
+
+			messages, text, system_msg, record, key_points = eval_result
+
+			# Final steps to get judgement
+			judgement = model.invoke(messages).content
+
+			if 'success' in judgement.lower().split('status:')[1]:  # This is the official criteria for success
+				evaluation = {'task_id': task_folder.name, 'judgement': judgement, 'success': True, 'error': None, 'score': 1.0}
+			else:  # This is the official criteria for failure
+				evaluation = {'task_id': task_folder.name, 'judgement': judgement, 'success': False, 'error': None, 'score': 0.0}
+
+			# Save the Online_Mind2Web_evaluation into the result.json file
+			result['Online_Mind2Web_evaluation'] = evaluation
+			with open(result_file, 'w') as f:
+				json.dump(result, f, indent=2)
+
+			return evaluation
+
+		except Exception as e:
+			return {'task_id': task_folder.name, 'judgement': None, 'success': False, 'error': str(e), 'score': 0.0}
+
+	except Exception as e:
+		return {'task_id': task_folder.name, 'judgement': None, 'success': False, 'error': str(e), 'score': 0.0}
+
+
+async def evaluate_all_saved_results(args) -> Dict:
+	"""
+	Evaluate all completed tasks in the saved_trajectories folder.
+
+	Returns:
+	    Dictionary containing evaluation summary
+	"""
+	trajectories_dir = Path('saved_trajectories')
+	if not trajectories_dir.exists():
+		return {'error': 'No saved trajectories found'}
+
+	# Define the model used as a judge
+	model = ChatOpenAI(
+		model='gpt-4o',
+		temperature=0.0,
+	)
+
+	# Create a semaphore to limit concurrent evaluations
+	semaphore = asyncio.Semaphore(args.parallel_evaluations)
+
+	async def evaluate_task(task_folder: Path) -> Dict:
+		async with semaphore:
+			judgement = await asyncio.to_thread(judge_task_result, model, task_folder)
+			logger.info(f'Completed evaluation for task {task_folder.name}. Result: {judgement["success"]}')
+			return judgement
+
+	# Get all task folders
+	task_folders = [f for f in trajectories_dir.iterdir() if f.is_dir()]
+
+	# Run evaluations in parallel
+	judgements = await asyncio.gather(*[evaluate_task(folder) for folder in task_folders])
+
+	# Calculate summary statistics
+	total_tasks = len(judgements)
+	successful_tasks = sum(1 for j in judgements if j['success'])
+	failed_tasks = sum(1 for j in judgements if not j['success'])
+	average_score = sum(j['score'] for j in judgements) / total_tasks if total_tasks > 0 else 0
+
+	summary = {
+		'timestamp': datetime.now().isoformat(),
+		'total_tasks': total_tasks,
+		'successful_tasks': successful_tasks,
+		'failed_tasks': failed_tasks,
+		'success_rate': successful_tasks / total_tasks if total_tasks > 0 else 0,
+		'average_score': average_score,
+		'detailed_results': judgements,
+	}
+
+	return summary
+
+
+async def run_multiple_tasks(
+	tasks: List[Task],
```

Comment:
```suggestion
	tasks: list[Task],
```

super minor but as of 3.12 and 3.13 the older imported `Dict[...]`, `List`, `Union` tupes are going to be phased out in favor of the newer variants: `dict[...]`, `list[...]`, `abc | def`